### PR TITLE
Orientjs v2.1.0 

### PR DESCRIPTION
**Pull request by saeedtabrizi**
case 1. Add a new type definition.
- [x] checked compilation succeeds with `--target es6` and `--noImplicitAny` options.
- [x] has correct [naming convention](http://definitelytyped.org/guides/contributing.html#naming-the-file)
- [x] has a [test file](http://definitelytyped.org/guides/contributing.html#tests) with the suffix of  `-tests.ts` or `-tests.tsx`.

case 2. Improvement to existing type definition.
- documentation or source code reference which provides context for the suggested changes.  url http://api.jquery.com/html .
  - it has been reviewed by a DefinitelyTyped member.

### CODE REVIEW

File: orientjs/orientjs.d.ts [hunk 1]
```diff
@@ -0,0 +1,500 @@
...
+        interface RecordMeta {
+            "@rid": string|RID;
+            "@version": string|number;
+           
+        }
+
+        interface Record {
+            rid: RID;
+            create(record: any, options?: any): Promise<Record> ;
+            get(record: Record & any, options?: any): Promise<Record | Buffer>;
```
Comment (mhegazy) [inline orientjs/orientjs.d.ts]: `any` subsumes any other type, so `Record & any` is just `any`, and so is `Record | any`. if `any` is allowed then just use `any` every where.

Comment (mhegazy) [inline orientjs/orientjs.d.ts]: this also applies to the other uses of `any` in this file.

File: orientjs/orientjs.d.ts [hunk 2]
```diff
@@ -0,0 +1,500 @@
...
+        interface Statement extends Query<any> {
+            select(param: string): Statement;
+            select(params: string[]): Statement;
+            traverse(param: string): Statement;
+            traverse(param: string[]): Statement;
+
+            strategy(param: string): Statement;
+            insert(param: string): Statement;
+            insert(param: string[]): Statement;
+            insert(param: any): Statement;
```
Comment (mhegazy) [inline orientjs/orientjs.d.ts]: same as my comment about `any` earlier. there is no reason to list these overloads if there is an `any` overload. just keep the `any` overload.

File: orientjs/orientjs.d.ts [hunk 3]
```diff
@@ -0,0 +1,500 @@
...
+            get(name: string, refresh?: boolean): Promise<Index>;
+            cacheData(indices: any[]): Promise<Db>;
+
+        }
+
+        interface Statement extends Query<any> {
+            select(param: string): Statement;
+            select(params: string[]): Statement;
+            traverse(param: string): Statement;
+            traverse(param: string[]): Statement;
```
Comment (mhegazy) [inline orientjs/orientjs.d.ts]: coolece the signatures as:

```ts
traverse(param: string[] | string): Statement;
```

Comment (mhegazy) [inline orientjs/orientjs.d.ts]: also applies to the other signatures like `select`, `insert`, etc..

File: orientjs/orientjs.d.ts [hunk 4]
```diff
@@ -0,0 +1,500 @@
...
+            near(latitudeProperty: string, longitudeProperty: string, longitude: number, latitude: number, maxDistanceInKms: number): Statement;
+            near(prop: any, maxDistanceInKms: number, longitude: number, latitude: number): Statement;
+
+            within(latitudeProperty: string, longitudeProperty: string, box: Array<Number>): Statement;
+
+            addParams(key: string, value: any): Statement;
+            addParams(value: any): Statement;
+            token(value: any): Statement;
+            buildStatement(): string;
+
```
Comment (mhegazy) [inline orientjs/orientjs.d.ts]: nit. extra white spaces.

File: orientjs/orientjs.d.ts [hunk 5]
```diff
@@ -0,0 +1,500 @@
...
+        }
+
+        interface Query<T> {
+
+            transform<T>(transformer: any): Query<T>;
+            transform<T>(transformer: Function): Query<T>;
```
Comment (mhegazy) [inline orientjs/orientjs.d.ts]: this signature will never be matched as overload resolution in TS goes in order of signatures, and the previous signature is general enough to match any valid call. so similar to my previous comments about `any`, do not include the last signature, it is not needed.

File: orientjs/orientjs.d.ts [hunk 6]
```diff
@@ -0,0 +1,500 @@
...
+            create(config: any): Promise<Db>;
+            drop(config: string): Promise<Db>;
+            drop(config: DbConfig): Promise<Db>;
+            list(): Promise<Db[]>;
+            exists(name: string, storageType?: string): Promise<boolean>;
+            freeze(name: string, storageType?: string): any;
+            release(name: string, storageType?: string): any;
+        }
+
+
```
Comment (mhegazy) [inline orientjs/orientjs.d.ts]: a lot of extra empty lines throughout the file.

**Comment by saeedtabrizi** [comment]
Hi @mhegazy 
as your notes i create a new PR https://github.com/DefinitelyTyped/DefinitelyTyped/pull/9813 

Thanks 
Saeed Tabrizi

**Pull request closed without merging**